subsys/net/lib/lwm2m/lwm2m_rd_client.c

```diff
@@ -261,6 +271,7 @@ static void sm_handle_failure_state(enum sm_engine_state sm_state)
 	} else if (client.engine_state == ENGINE_DEREGISTER_SENT) {
 		event = LWM2M_RD_CLIENT_EVENT_DEREGISTER_FAILURE;
 	}
+	k_mutex_unlock(&state_lock);
```

Comment:
There is an early `return` within the locked context.

Reply:
thanks